dataset/include/scipp/dataset/bucket.h

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include "scipp/dataset/dataset.h"
+
+namespace scipp::dataset::buckets {
+
+[[nodiscard]] SCIPP_DATASET_EXPORT Variable
+concatenate(const VariableConstView &var0, const VariableConstView &var1);
```

Comment:
You are right that *parts* of this could be moved into `variable`, and I may eventually do so. At this point though we are mainly using this with buckets for data arrays or datasets, so the related code has to be here. I think we will anyway end up moving things around once we better understand how the developing toolbox for buckets will look like. At this point it is still very unclear which features we will support and how.